chore(docs): docs for devnet.20251212

critesjosh opened the pull request:
## Summary

- Adds versioned documentation for `v3.0.0-devnet.20251212`
- Removes the old `v3.0.0-devnet.5` documentation version
- Sets up redirects from old devnet.5 URLs to the new devnet.20251212 version
- Updates CLI reference generation and documentation links
- Updates tutorials and foundational topics with latest changes from next branch
- Fixes devnet label format in docusaurus config
- Updates devnet version and URLs in getting started guide

alejoamiras reviewed (approved):
approving in name of @ciaranightingale

critesjosh committed:
chore(docs): docs for devnet.20251212

## Summary

- Adds versioned documentation for `v3.0.0-devnet.20251212`
- Removes the old `v3.0.0-devnet.5` documentation version
- Sets up redirects from old devnet.5 URLs to the new devnet.20251212 version
- Updates CLI reference generation and documentation links
- Updates tutorials and foundational topics with latest changes from next branch
- Fixes devnet label format in docusaurus config
- Updates devnet version and URLs in getting started guide

Co-authored-by: Josh Crites <jc@joshcrites.com>
Co-authored-by: ludamad <domuradical@gmail.com>
